framework/src/modules/chain/transport/transport.js

```diff
@@ -390,28 +424,79 @@ class Transport {
 	}
 
 	/**
-	 * Description of postTransactions.
+	 * Description of postTransactionsAnnouncement.
 	 *
 	 * @todo Add @param tags
 	 * @todo Add @returns tag
 	 * @todo Add description of the function
 	 */
-	async postTransactions(query) {
+	async postTransactionsAnnouncement({ data, peerId }) {
 		if (!this.constants.broadcasts.active) {
 			return this.logger.debug(
 				'Receiving transactions disabled by user through config.json',
 			);
 		}
 
-		const errors = validator.validate(definitions.WSTransactionsRequest, query);
+		const errors = validator.validate(definitions.WSTransactionsRequest, data);
+
+		if (errors.length) {
+			this.logger.debug({ err: errors }, 'Invalid transactions body');
+			// TODO: If there is an error, invoke the applyPenalty action on the Network module once it is implemented.
+			throw errors;
+		}
+
+		const unknownTransactions = await this.checkTransactionsIDs(data);
```

Comment:
Prefer using more descriptive name like `filterUnknownTransactionIds` or something instead of `checkTransactionsIDs`